### PR TITLE
[UTC] Don't leave dangling CHECK-SAME when removing CHECK lines

### LINKED ISSUE #78517: [UTC] Update test check removes check lines for declarations (and leaves the argument checks)
Since da28c3309413475146fdeb9194e44b7a63b7484e the check lines of certain declarations are removed but the argument check-same lines are kept. That is obviously broken.

As an example, this is what happens if I update the Attributor tests now:
```
  --- a/llvm/test/Transforms/Attributor/misc_crash.ll
  +++ b/llvm/test/Transforms/Attributor/misc_crash.ll
  @@ -124,7 +124,6 @@ define internal i16 @bar3(ptr %p1, i16 %p2) {
     ret i16 %p2
   }

  -; CHECK-LABEL: declare {{[^@]+}}@func6
   ; CHECK-SAME: (ptr)
   declare void @func6(ptr)
   ;.
~
```

### PR DESCRIPTION
**Pull request by hnrklssn**
When removing only lines that are global value CHECK lines, a related CHECK-SAME line could be left dangling without a previous line to belong to.

Resolves #78517

### CODE REVIEW

File: llvm/utils/UpdateTestChecks/common.py [hunk 1]
```diff
@@ -388,7 +388,11 @@ def itertests(
 
 
 def should_add_line_to_output(
-    input_line, prefix_set, skip_global_checks=False, comment_marker=";"
+    input_line,
+    prefix_set,
+    skip_global_checks=False,
+    skip_same_checks=False,
+    comment_marker=";",
```
Comment (arichardson) [inline llvm/utils/UpdateTestChecks/common.py:395]:
```suggestion
    input_line,
    prefix_set,
    *,
    skip_global_checks=False,
    skip_same_checks=False,
    comment_marker=";",
```

Since you added a new parameter it seems like no callers actually pass `comment_marker` here?

I find function calls with various boolean flags quite hard to parse, maybe make these parameters named arguments?

File: llvm/utils/update_test_checks.py [hunk 2]
```diff
@@ -235,6 +235,7 @@ def main():
             )
         else:
             # "Normal" mode.
+            skip_same_check_line = False
```
Comment (arichardson) [inline llvm/utils/update_test_checks.py:238]:
```suggestion
            dropped_previous_line = False
```
I think this name makes it easier to understand.

File: llvm/utils/UpdateTestChecks/common.py (hunk 1 above)
Comment (hnrklssn) [reply llvm/utils/UpdateTestChecks/common.py:395]: Yeah, the call in `update_test_checks.py` is the only one passing anything other than `input_line` and `prefix_set`. The comment marker is always `";"` currently.

**Review by arichardson — APPROVED** [review]
LGTM assuming all tests are passing (do the other update scripts need changes too, e.g. update_cc_test_checks.py?)

**Comment by hnrklssn** [comment]
> LGTM assuming all tests are passing (do the other update scripts need changes too, e.g. update_cc_test_checks.py?)

No, doesn't seem like it, luckily
